cluster-autoscaler/simulator/drainability/rules/rules_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rules
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/drainability"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/drain"
+)
+
+func TestDrainable(t *testing.T) {
+	for desc, tc := range map[string]struct {
+		rules Rules
+		want  drainability.Status
+	}{
+		"no rules": {
+			want: drainability.NewUndefinedStatus(),
+		},
+		"first non-undefined rule returned": {
+			rules: Rules{
+				fakeRule{drainability.NewUndefinedStatus()},
+				fakeRule{drainability.NewDrainableStatus()},
+				fakeRule{drainability.NewSkipStatus()},
+			},
+			want: drainability.NewDrainableStatus(),
+		},
+		"override match": {
+			rules: Rules{
+				fakeRule{drainability.Status{
+					Outcome:   drainability.DrainOk,
+					Overrides: []drainability.OutcomeType{drainability.BlockDrain},
+				}},
+				fakeRule{drainability.NewBlockedStatus(drain.NotEnoughPdb, nil)},
+			},
+			want: drainability.Status{
+				Outcome:   drainability.DrainOk,
+				Overrides: []drainability.OutcomeType{drainability.BlockDrain},
+			},
+		},
+		"override no match": {
+			rules: Rules{
+				fakeRule{drainability.Status{
```

Comment:
nit: How about `drainability.NewDrainableStatus(drainability.OverridingOnly(drainability.SkipDrain))`? Or is it not really easier to read?